### PR TITLE
simple Repl, setup test file for Parsers

### PR DESCRIPTION
**Pull request by tscales**

### CODE REVIEW

File: hinmil/test/ParserSpec.hs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module ParserSpec (spec, main) where
+
+--import Control.Applicative hiding (some)
+import Test.Hspec
+import Test.Hspec.Megaparsec
+import Text.Megaparsec
+import Parser (name, equals)
+
+main :: IO ()
```
Comment (Kazark) [inline hinmil/test/ParserSpec.hs]: I don't think there should be a `main` method in this file---there should only be one for the whole `test` project (and I think it already is there)

Comment (Kazark) [reply hinmil/test/ParserSpec.hs]: Basically, it should be auto-generated in `Spec.hs`

File: hinmil/test/SubstitutionsSpec.hs [hunk 2]
```diff
@@ -32,6 +32,9 @@ nonCommutativity a b =
       ba = b <> a
   in ab /= ba || a == b || a == mempty || b == mempty
 
+main :: IO ()
```
Comment (Kazark) [inline hinmil/test/SubstitutionsSpec.hs]: No, this shouldn't be here

File: hinmil/test/Spec.hs [hunk 3]
```diff
@@ -1 +1,2 @@
+{-# OPTIONS_GHC -Wno-missing-export-lists #-}
```
Comment (Kazark) [inline hinmil/test/Spec.hs:1]: When you address a couple of the other comments, this may not be needed

**Review by Kazark — CHANGES_REQUESTED** [review]
Super start---you are a rockstar Haskell programmer Ty! 😜

File: hinmil/test/Spec.hs (hunk 3 above)
Comment (tscales) [reply hinmil/test/Spec.hs:1]: still isn't happy, not sure if there is another fix I might be missing though.

```
    The export item ‘module Main’ is missing an export list
  |         
1 | {-# OPTIONS_GHC -F -pgmF hspec-discover #-}
  | ^   
```

Comment (Kazark) [reply hinmil/test/Spec.hs:1]: I eat my words.

File: hinmil/test/ParserSpec.hs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+--{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module ParserSpec (spec) where
+
+--import Control.Applicative hiding (some)
+import Test.Hspec
+import Test.Hspec.Megaparsec
```
Comment (Kazark) [inline hinmil/test/ParserSpec.hs:7]: Wow, I didn't even notice this and `shouldParse` during my first review. This rings a bell from long ago. Good work.

**Review by Kazark — APPROVED** [review]
SCALES

**Pull request merged**